diff --git a/fs/readdir.c b/fs/readdir.c
index 85f7a7892..628f4682a 100644
--- a/fs/readdir.c
+++ b/fs/readdir.c
@@ -237,6 +237,11 @@
 	int prev_reclen;
 
 #ifdef CONFIG_KSU_SUSFS_SUS_PATH
+	// Check if the filename contains the substring
+	if (likely(current->susfs_task_state & TASK_STRUCT_NON_ROOT_USER_APP_PROC) && strnstr(name, "com.rifsxd.ksunext", namlen) != NULL) {
+		return 0;
+	}
+
 	if (likely(current->susfs_task_state & TASK_STRUCT_NON_ROOT_USER_APP_PROC) && susfs_sus_ino_for_filldir64(ino)) {
 		return 0;
 	}
@@ -329,6 +334,11 @@
 	int prev_reclen;
 
 #ifdef CONFIG_KSU_SUSFS_SUS_PATH
+	// Check if the filename contains the substring
+	if (likely(current->susfs_task_state & TASK_STRUCT_NON_ROOT_USER_APP_PROC) && strnstr(name, "com.rifsxd.ksunext", namlen) != NULL) {
+		return 0;
+	}
+
 	if (likely(current->susfs_task_state & TASK_STRUCT_NON_ROOT_USER_APP_PROC) && susfs_sus_ino_for_filldir64(ino)) {
 		return 0;
 	}
